arangod/Cluster/ServerState.cpp

```diff
@@ -632,6 +644,34 @@ bool ServerState::checkEngineEquality(AgencyComm& comm) {
   return true;
 }
 
+/// @brief check equality of naming conventions settings with other registered servers
+bool ServerState::checkNamingConventionsEquality(AgencyComm& comm) {
+  // our own setting
+  bool const extendedNamesForDatabases = _server.getFeature<DatabaseFeature>().extendedNamesForDatabases();
+
+  AgencyCommResult result = comm.getValues(currentServersRegisteredPref);
+  if (result.successful()) {  // no error if we cannot reach agency directly
+
+    auto slicePath = AgencyCommHelper::slicePath(currentServersRegisteredPref);
+    VPackSlice servers = result.slice()[0].get(slicePath);
+    if (!servers.isObject()) {
+      return true;  // do not do anything harsh here
+    }
+
+    for (VPackObjectIterator::ObjectPair pair : VPackObjectIterator(servers)) {
+      if (pair.value.isObject()) {
+        VPackSlice setting = pair.value.get(::extendedNamesDatabasesKey);
+        if (setting.isBool() && setting.getBool() != extendedNamesForDatabases) {
+          // different setting detected
+          return false;
```

Comment:
I will adjust this now to 
* abort the startup with a fatal error
* print the cluster servers' configuration w.r.t. to the setting
However, I can't get this to print the server endpoints, probably because at that point during the startup the ClusterInfo isn't yet fully initialized and knows about the server-to-endpoint mappings.